SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -102,6 +105,70 @@ public static ProjectInfoAnalysisResult GenerateFile(AnalysisConfig config, ILog
             return result;
         }
 
+        /// Appends the sonar.projectBaseDir value. This is calculated as follows:
+        /// 1. the user supplied value, or if none
+        /// 2. the sources directory if running from TFS Build or XAML Build, or
+        /// 3. the common root path of projects, or if there isn't any
+        /// 4. the .sonarqube/out directory
```

Comment:
I don't understand the last point. .sonarqube/out should be relative to the base dir, no?